File: CharityWebBackend/src/main/java/group3/project/charityweb/service/InteractionService.java
package group3.project.charityweb.service;

import group3.project.charityweb.model.dto.request.InteractionRequest;
import group3.project.charityweb.model.dto.response.InteractionResponse;
import org.springframework.data.domain.Page;

public interface InteractionService {
    String createInteraction(String username, String activityId, InteractionRequest request);
    Page<InteractionResponse> getActivityInteractions(String activityId, int page, int size);
    void deleteInteraction(String username, String interactionId);
}

File: CharityWebFrontend/src/services/InteractionService.js
import api from './api';

export const createInteraction = (activityId, data) =>
    api.post(`/activities/${activityId}/interactions`, data);

export const getActivityInteractions = (activityId) =>
    api.get(`/activities/${activityId}/interactions`);

export const deleteInteraction = (interactionId) =>
    api.delete(`/interactions/${interactionId}`);
